syft/pkg/cataloger/rust/cargo_metadata.go

```diff
@@ -1,11 +1,19 @@
 package rust
 
-import "github.com/anchore/syft/syft/pkg"
+import (
+	"reflect"
+
+	"github.com/anchore/syft/syft/pkg"
+)
 
 type CargoMetadata struct {
 	Packages []pkg.CargoPackageMetadata `toml:"package"`
 }
 
+func init() {
```

Comment:
is this `init()` function the right place to do something like this?

Reply:
Spoke with @luhring and it seems like we should probably move `cargo_metadata.go` up to `syft/pkg`; I'll make this change.

Reply:
why so? This cargo metadata is meant for parsing a cargo metadata file and it's `pkg.CargoPackageMetadata` that resides on the `pkg.Package.Metadata` itself

Reply:
@wagoodman this would be consistent with all the other package metadata types being defined in [syft/pkg](https://github.com/anchore/syft/tree/main/syft/pkg). But I'm happy to hear other options; maybe move the `pkg.CargoPackageMetadata` definition here?

Reply:
I see the confusion: we're in agreement about the fact that we shouldn't be using `rust.CargoMetadata{}` here. What I was confused at is that there is already cargo metadata up in `pkg` (https://github.com/anchore/syft/blob/main/syft/pkg/cargo_package_metadata.go#L10), it looks like this was left behind mistakenly. We should indeed switch to using the metadata that is up in pkg (as was originally intended).

Reply:
I think the right thing to do here is move the `cargo_metadata.go`, I've done this -- please let me know if I am misunderstanding